zh-cn/yodaos-source/framework/01-application.md

```diff
@@ -0,0 +1,186 @@
+
+```js
+class MyComponent {
+  constructor (runtime) {
+    this._runtime = runtime
+    // do what you want to
+  }
+}
+module.exports = MyComponent
+```
+
+YodaOS Component 开发者通过类构造函数的第一个参数访问到 `runtime`，并通过它可以调用到其他组件的方法和事件，以及其他应用框架服务（AFS）的基础功能。
+
+> 所有组件都是运行在 [AFS][] 内部，因此在编写组件时需要是否小心内存分配和性能。
+
+#### YodaOS Descriptor
+
+YodaOS Descriptor 是 [AFS][] 提供的应用接口定义机制，它是一种基于 JavaScript/JSON 的 DSL，用于定义 [YodaOS Application][] 的核心接口。
+
+与 [YodaOS Component][] 类似，[AFS][] 会在启动时新建 [YodaOS Descriptor][] 的加载器，不过现在还不提供可配置的入口，因此所有的核心接口都需要在 [`runtime/descriptor`](https://github.com/yodaos-project/yodart/tree/master/runtime/descriptor) 定义。
+
+在 YodaOS 中，新增一个 Descriptor 与 [YodaOS Component][] 类似：
+
+```js
+// runtime/descriptor/foobar.js
+var Descriptor = require('../lib/descriptor')
+
+class MyDescriptor extends Descriptor {
+  constructor (runtime) {
+    super(runtime, 'foobar')
+  }
+  hello () {
+    return Promise.resolve('hello')
+  }
+}
+MyDescriptor.values = {
+  some: {}
+}
+MyDescriptor.events = {
+  created: {}
+}
+MyDescriptor.methods = {
+  hello: {
+    returns: 'promise'
+  }
+}
+
+module.exports = MyDescriptor
+```
+
+上面的代码描述了一个 [YodaOS Descriptor][] 的基本元素：
+
+- 继承自 [Descriptor](https://github.com/yodaos-project/yodart/tree/master/runtime/lib/descriptor.js) 类，它定义在 `runtime/lib/descriptor.js`
+- 在类构造函数中，调用 `super(runtime, namespace)` 并设置该接口类的命名空间（namespace）
+- 在 `MyDescriptor.values` 定义：使用该类时可访问的值
+- 在 `MyDescriptor.events` 定义：使用该类时的事件列表
+- 在 `MyDescriptor.methods` 定义：使用该类时的所有方法
+
+到此即完成了 [AFS][] 端的工作，接下来介绍如何在客户端（应用）访问定义的接口：
+
+```js
+class MyApplication {
+  constructor () {
+    // corresponding to the namespace
+    this.api = global[Symbol.for('yoda#foobar')].foobar
+    this.api.hello()
+  }
+}
+```
+
+在实际的应用开发中，其实并不需要像上面这样写，在 [`@yodaos/application`](https://github.com/yodaos-project/yodart/tree/master/packages/%40yodaos/application) 包中已经为开发者封装好了更为便捷的使用方式，这里的例子仅作为了解 [YodaOS Descriptor][] 所用。
+
+#### 应用的启动
+
+你可以把 [YodaOS][] 理解为一个浏览器，每当用户进行交互时，就相当于在浏览器中打开不同的 URL，因此对于每个应用程序来说，它的入口就是处理不同的 URL 请求，因此这里就涉及到两个过程：
+
+- 其他系统服务发送 URL 到 [AFS][]（比如 [Voice Interface Driver][] 收到一个云端指令）
+- [AFS][] 解析 URL，并将请求分发到对应的应用（若应用未启动，则需要先初始化应用）
+
+所以，工作中的 YodaOS 其实就是一个在不断处理 URL 的浏览器，不过生成 URL 的方式不再需要用户去手动输入。
+
+> 对于想要了解应用分发机制的读者，可以阅读 [runtime/app-runtime.js#L261-L281](https://github.com/yodaos-project/yodart/blob/v8.2.x/runtime/app-runtime.js#L261-L281)。
+
+接下来，我们更多地了解下应用在启动时的细节。
+
+YodaOS 将应用分为两种运行模式：`extapp` 与 `light`，前者以独立进程的方式运行，而后者即运行在 [AFS][] 进程。`light` 模式主要是考虑到在语音交互场景下，有一些应用逻辑十分简单（如播放音频），为了减少用户响应的时延以及进程开销，从而引入了 `light` 模式。
+
+> 对于应用开发者来说，并不需要关心应用模式，它只能由系统配置文件 [`/etc/yoda/app-loader-config.json`](https://github.com/yodaos-project/yodart/blob/v8.2.x/etc/yoda/app-loader-config.json#L5-L9) 来定义。
+
+确定了应用的运行模式，我们就可以了解应用的启动机制了，可以在 [`runtime/app`](https://github.com/yodaos-project/yodart/blob/master/runtime/app) 目录下发现以下文件：
+
+- `default-launcher.js` 默认的应用启动方式，通过 `child_process.fork()` 来启动进程
+- `executable-launcher.js` 与前者不同，该启动器用于支持 C/C++ 应用
```

Comment:
```suggestion
- `executable-launcher.js` 与前者不同，该启动器用于支持任意可执行文件
```

Reply:
I'm strongly recommend to emphasize the C/C++ application here:

```
该启动器用于支持可执行程序作为入口程序，它被用于提供一些非 JavaScript 应用开发环境，如 C/C++、Rust、Golang 等。
```